synapse/storage/databases/main/receipts.py

```diff
@@ -113,24 +113,6 @@ def __init__(
             prefilled_cache=receipts_stream_prefill,
         )
 
-        self.db_pool.updates.register_background_index_update(
```

Comment:
Why move these to the non-worker store?

Reply:
I moved them to the `*BackgroundUpdateStore`. I thought that's where we usually put the background updates?
Is there a motivation for having these on the worker store that I've completely missed?

Reply:
There's no motivation besides it being consistent with other examples I saw. 🤷